File: gc_lang/fr/modules/phonet.tpl.py
# -*- encoding: UTF-8 -*-
# Grammalecte - Suggestion phonétique
# License: GPL 3

import re


## generated data

_dWord = ${dWord}

_lSet = ${lSet}

## end of generated data


def hasSimil (sWord):
    "return True if there is list of words phonetically similar to sWord"
    if not sWord:
        return False
    if sWord[0:1].isupper():
        sWord = sWord.lower()
    if sWord in _dWord:
        return True
    return False


def getSimil (sWord):
    "return list of words phonetically similar to sWord"
    if not sWord:
        return []
    if sWord[0:1].isupper():
        sWord = sWord.lower()
    if sWord in _dWord:
        return _lSet[_dWord[sWord]]
    return []

File: gc_lang/fr/modules-js/phonet.tpl.js
// Grammalecte - Suggestion phonétique

// generated data

const _dWord = new Map (${dWord});

const _lSet = ${lSet};

// end of generated data

function hasSimil (sWord) {
    // return True if there is list of words phonetically similar to sWord
    if (!sWord) {
        return false;
    }
    if (sWord.slice(0,1)._isUpperCase()) {
        sWord = sWord.toLowerCase();
    }
    return _dWord.has(sWord);
}

function getSimil (sWord) {
    // return list of words phonetically similar to sWord
    if (!sWord) {
        return [];
    }
    if (sWord.slice(0,1)._isUpperCase()) {
        sWord = sWord.toLowerCase();
    }
    if (_dWord.has(sWord)) {
        return _lSet[_dWord.get(sWord)];
    }
    return [];
}

exports.hasSimil = hasSimil;
exports.getSimil = getSimil;
